drivers/sensor/wsen_hids_2525020210002/wsen_hids_2525020210002.c

```diff
@@ -0,0 +1,248 @@
+{
+	struct hids_2525020210002_data *data = dev->data;
+
+	if (precision->val1 < hids_2525020210002_precision_Low ||
+	    precision->val1 > hids_2525020210002_precision_High || precision->val2 != 0) {
+		LOG_ERR("Bad precision configuration %d", precision->val1);
+		return -EINVAL;
+	}
+
+	data->sensor_precision = (hids_2525020210002_precision_t)precision->val1;
+
+	return 0;
+}
+
+/* Set heater option. */
+static int hids_2525020210002_heater_set(const struct device *dev,
+					 const struct sensor_value *heater)
+{
+	struct hids_2525020210002_data *data = dev->data;
+
+	if (heater->val1 < hids_2525020210002_heater_Off ||
+	    heater->val1 > hids_2525020210002_heater_On_20mW_100ms || heater->val2 != 0) {
+		LOG_ERR("Bad heater option %d", heater->val1);
+		return -EINVAL;
+	}
+
+	data->sensor_heater = (hids_2525020210002_heater_t)heater->val1;
+
+	return 0;
+}
+
+static int hids_2525020210002_attr_set(const struct device *dev, enum sensor_channel chan,
+				       enum sensor_attribute attr, const struct sensor_value *val)
+{
+
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_set() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_PRECISION:
+		return hids_2525020210002_precision_set(dev, val);
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_HEATER:
+		return hids_2525020210002_heater_set(dev, val);
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int hids_2525020210002_attr_get(const struct device *dev, enum sensor_channel chan,
+				       enum sensor_attribute attr, struct sensor_value *val)
+{
+	struct hids_2525020210002_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_ALL) {
+		LOG_WRN("attr_get() is not supported on channel %d.", chan);
+		return -ENOTSUP;
+	}
+
+	if (val == NULL) {
+		LOG_WRN("address of passed value is NULL.");
+		return -EFAULT;
+	}
+
+	switch ((int)attr) {
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_PRECISION:
+		val->val1 = data->sensor_precision;
+		val->val2 = 0;
+		break;
+	case SENSOR_ATTR_WSEN_HIDS_2525020210002_HEATER:
+		val->val1 = data->sensor_heater;
+		val->val2 = 0;
+		break;
+	default:
+		LOG_ERR("Operation not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static const struct sensor_driver_api hids_2525020210002_driver_api = {
+	.attr_set = hids_2525020210002_attr_set,
+	.attr_get = hids_2525020210002_attr_get,
+	.sample_fetch = hids_2525020210002_sample_fetch,
+	.channel_get = hids_2525020210002_channel_get,
+};
+
+static int hids_2525020210002_init(const struct device *dev)
+{
+	const struct hids_2525020210002_config *const config = dev->config;
+	struct hids_2525020210002_data *data = dev->data;
+	struct sensor_value precision, heater;
+
+	/* Initialize WE sensor interface */
+	HIDS_Get_Default_Interface(&data->sensor_interface);
+	data->sensor_interface.interfaceType = WE_i2c;
+	data->sensor_interface.handle = (void *)&config->bus_cfg.i2c;
```

Comment:
Need to check if the I2C bus is ready with  `i2c_is_ready_dt` before using it.

Reply:
For every driver now the correct is ready function is called depending on the interface